src/components/Weather/Weather.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from "react";
+import { API_KEY } from "./Weather.constant";
+import "./Weather.css";
+import axios from "axios";
+import { useIntl } from "react-intl";
+
+type Main = {
+  temp: number;
+};
+
+type Sys = {
+  sunrise: number;
+  sunset: number;
+};
+
+type WeatherResponse = {
+  main: Main;
+  sys: Sys;
+  name: string;
+};
+
+function Weather() {
+  const [lat, setLat] = useState<number | null>(null);
+  const [long, setLong] = useState<number | null>(null);
+  const [data, setData] = useState<WeatherResponse | null>(null);
```

Comment:
please, do not use abbreviations. use more meaningful name than data

Reply:
done